### DISCUSSION TITLE
background color transition doesn't work

### DISCUSSION THREAD

**Discussion by alephpi** [vuejs/vitepress#2517] [Q&A]
Hi, here is a [example](https://stackblitz.com/edit/vite-r5o2q1?file=docs%2Findex.md).

Basically I just want to realize the background color transition like vuejs official doc. (thought it uses vitepress theme, there is a slight difference), I simply add one line of css, but doesn't seem to work.

**Comment by Liberty-liu** [comment]
[https://github.com/vuejs/docs/blob/main/.vitepress/theme/components/Home.vue#L115](https://github.com/vuejs/docs/blob/main/.vitepress/theme/components/Home.vue#L115)

**Reply by brc-dd** [reply]
IG the OP was referring to adding transition to the background (like animate it when the user changes dark/light mode). But that is blocked by us intentionally:

https://github.com/vuejs/vitepress/blob/c4909e4298ec706cf1762cb36af03e5fd3637ccc/src/client/theme-default/components/VPSwitchAppearance.vue#L45-L67

Refer this: https://paco.me/writing/disable-theme-transitions

To disable that, you'd need to patch the file: https://vitepress.dev/guide/extending-default-theme#overriding-internal-components
